File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java
class Solution {
    public int findDuplicate(int[] nums) {
        int tortoise = nums[0];
        int hare = nums[tortoise];

        while (tortoise != hare) {
            tortoise = nums[tortoise];
            hare = nums[nums[hare]];
        }

        tortoise = 0;

        while (tortoise != hare) {
            tortoise = nums[tortoise];
            hare = nums[hare];
        }

        return hare;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        all_nums = set(nums)
        visited = set()
        best = 0
        for num in all_nums:
            if num in visited:
                continue
            else:
                start = num
                while num in all_nums:
                    visited.add(num)
                    num += 1
                
                best = max(best, num - start)
        
        return best
